modules/oe_content_timeline_field/tests/modules/oe_timeline_test_constraint/src/Plugin/Validation/Constraint/TestConstraintValidator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_timeline_test_constraint\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates required fields for Timeline paragraph.
+ */
+class TestConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    if ($value->isEmpty()) {
+      return;
+    }
+
+    $parameters = [
+      '%label' => 'label',
+      '%title' => 'title',
+      '%body' => 'body',
+    ];
+    foreach ($value as $delta => $item) {
+      /** @var \Drupal\oe_content_timeline_field\Plugin\Field\FieldType\TimelineFieldItem $item */
+      $value = $item->getValue();
+      if ($value['label'] === '' || $value['label'] === NULL) {
```

Comment:
Could be single 'If' statement and better if first will be checking of body value.